feat: code refactor

sfeSantos opened the pull request:
This should be tested. Do you plan to use mockito or something else for tests? And still has room for improvements

sfeSantos committed:
feat: code refactor

CedricNanni commented:
I was not planning to use mockito but I am open. Any suggestion?. Obviously testing multiple RDMS requires some infrastructure. I will test the PR against our infrastructure. It is a detail but don't you think that the SequenceInfo should be private and may called sequence metadata?

sfeSantos committed:
feat: Change of visibility and renaming

sfeSantos commented:
I renamed and changed the visibility. Related to the test infrastructure maybe [Spock ](https://spockframework.org)would be a good idea for testing. Tell me if the changes are still working and what do you think about Spock

CedricNanni commented:
Thanks. I am gonna to test the changes today. I don't know Spock but I will have look and see how we can use it.

CedricNanni commented:
It passed all tests. Well done!

sfeSantos commented:
Glad to hear. I think this approach can be followed for the rest of the classes related to metadata

CedricNanni commented:
I agree.

CedricNanni commented:
Just that you are aware of on what I am working on. I am refactoring SelectQueryBuilder to add UpdateQueryBuilder and DeleteQueryBuilder. I think I will commit it today in main branch.

sfeSantos commented:
Nice, I'll keep looking in the code to see if I can improve something else. But I'm working in the same time on my project in Rust (trying to learn it) [mockiapi](https://github.com/sfeSantos/mockiapi) and I also created a create for it [crate](https://crates.io/search?q=mockiapi)

CedricNanni commented:
No worries, it is the same for me. In fact I am migrating in parallel the code of one our project to norm4j and it is quite time consuming despite being close to JPA. I will need to add from time to time new features to norm4j to ease the migration.

sfeSantos commented:
When u have time give a look into my project and let me know yout thoughts.

On Sun, Apr 6, 2025 at 11:52 AM Cedric Nanni ***@***.***>
wrote:

> No worries, it is the same for me. In fact I am migrating in parallel the
> code of one our project to norm4j and it is quite time consuming despite
> being close to JPA. I will need to add from time to time new features to
> norm4j to ease the migration.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aprilsoftware/norm4j/pull/4#issuecomment-2781337669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BO4CWJGDKSN6ZMVWQ5TYNL32YD2PFAVCNFSM6AAAAAB2PFJJTCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOBRGMZTONRWHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: CedricNanni]*CedricNanni* left a comment (aprilsoftware/norm4j#4)
> <https://github.com/aprilsoftware/norm4j/pull/4#issuecomment-2781337669>
>
> No worries, it is the same for me. In fact I am migrating in parallel the
> code of one our project to norm4j and it is quite time consuming despite
> being close to JPA. I will need to add from time to time new features to
> norm4j to ease the migration.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aprilsoftware/norm4j/pull/4#issuecomment-2781337669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BO4CWJGDKSN6ZMVWQ5TYNL32YD2PFAVCNFSM6AAAAAB2PFJJTCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOBRGMZTONRWHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Eduardo

CedricNanni commented:
I will.
